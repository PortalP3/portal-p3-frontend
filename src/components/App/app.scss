@import "src/utilities/_variables";
@import url('https://fonts.googleapis.com/css?family=Roboto:100,700');

body {
  margin: 0;
  background: #d2d3c6;
  font-family: $font_family;
}

div.app {
  display: flex;
  justify-content: center;
  flex-direction: column;
}
