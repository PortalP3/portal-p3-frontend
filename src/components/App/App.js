import React from 'react'

import Header from '../Header/Header'
import Main from '../Main/Main'
import Footer from '../Footer/Footer'

import './app.scss'

const App = () => {
  return (
    <div className="app">
      <Header />
      <Main />
      <Footer />
    </div>
  )
}

export default App
