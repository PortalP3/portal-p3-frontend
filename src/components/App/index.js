import React, {Component} from 'react'
import PropTypes from 'prop-types'

import WordpressClient from '../../clients/WordpressClient'

import './app.scss'

export default class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      categories: []
    }
    this.wordpressClient = this.props.wordpressClient
  }

  componentDidMount() {
    this.wordpressClient.getCategories().then((categories) => {
      this.setState({categories: categories})
    })
  }

  render() {
    console.log(this.state.categories)

    return (
      <div className="app">
        Hello P3 World!
        <br /><br />
        {this.state.categories.data ? this.state.categories.data[0].name : ''}
      </div>
    );
  }
}

App.defaultProps = {
  wordpressClient: new WordpressClient()
}

App.propTypes = {
  wordpressClient: PropTypes.shape()
}
