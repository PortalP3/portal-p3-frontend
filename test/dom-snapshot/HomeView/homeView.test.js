import React from 'react'
import renderer from 'react-test-renderer';

import App from '../../../src/components/App'

test('render Home view', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
