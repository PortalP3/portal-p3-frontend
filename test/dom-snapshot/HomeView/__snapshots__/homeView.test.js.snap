// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Home view 1`] = `
<div
  className="app"
>
  <header>
    <div
      className="logo-portal-info-container"
    >
      <div
        className="logo-portal"
      >
        PORTAL P3
        <p>
          Esta plataforma tiene como objetivo el compartir conocimiento crítico que pueda alimentar nuestro compromiso por la justicia social y económica
        </p>
      </div>
    </div>
  </header>
  <div
    className="main"
  >
    <h1>
      TEMÁTICAS
    </h1>
    <div
      className="category-container"
    />
  </div>
  <footer>
    <p>
      Portal P3 - Abacaxi - Copyrights 2018
    </p>
  </footer>
</div>
`;
